app/components/Icon.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import Svg,{
+    Circle,
+    G,
+    Line,
+    Path,
+    Polygon,
+    Polyline,
+    Rect,
+} from 'react-native-svg';
+
+const IconMap = {
+  "hamburger": Hamburger,
+  "chevron-up": ChevronUp
+}
+
+export default function Icon (props) {
+  let { name, style, fill = "black", size = 16, onClick, ...rest } = props;
+  const Component = IconMap[props.name] ? IconMap[props.name] : null;
+
+  return (
+    <Svg
+      width={size}
+      height={size}
+      style={style}
+      viewBox="0 0 16 16"
+      onClick={onClick}
+    >
+      <Component {...rest} fill={fill} />
+    </Svg>
+  )
+}
+
+function Hamburger(props) {
```

Comment:
I don't have a strong preference, but I wonder if it'd be more declarative to export the `Hamburger` / `ChevronUp` components and then other components would be able to reference them like `<HamburgerIcon fill="..." />` rather than `<Icon name="hamburger" fill="..." />`